packages/ember-extension-support/lib/models/inspected_node.js

```diff
@@ -0,0 +1,234 @@
+import Component from "ember-views/views/component";
+import { A as emberA } from "ember-runtime/system/native_array";
+import { get } from "ember-metal/property_get";
+
+/**
+ * Wraps a render node to provide debugging API for external services
+ * such as the Ember Inspector.
+ *
+ * Ideally public methods on this class should completely hide
+ * the internal implementations of render nodes.
+ *
+ * @class InspectedNode
+ * @param {Object} renderNode The Htmlbars render node
+ * @param {Object} parent The parent InspectedNode instance
+ */
+function InspectedNode(renderNode, parent) {
+  this.renderNode = renderNode;
+  this.parent = parent;
+}
+
+InspectedNode.prototype =  {
+  /**
+   * Useful for subclassing.
+   *
+   * @property constructor
+   * @type {Function}
+   */
+  constructor: InspectedNode,
+
+  /**
+   * The render node we are representing and adding
+   * debugging functionality to.
+   *
+   * This property is set upon creation of the instance.
+   *
+   * @property renderNode
+   * @type {Object}
+   * @private
+   */
+  renderNode: null,
```

Comment:
This is internal and should not be leaked, it should at least have an underscore prefix and not be documented like it is a thing.